sips/sip-45.md

```diff
@@ -0,0 +1,78 @@
+| Description         | Enable effective prioritized transaction submission. |
+| Author              | Shio Coder <shio.coder@gmail.com> |
+| Editor              | Will Riches <will@sui.io, @wriches> |
+| Type                | Standard |
+| Category            | Core |
+| Created             | 2024-11-26 |
+| Comments-URI        | https://sips.sui.io/comments-45 |
+| Status              | Fast Track |
+| Requires            | N/A |
+
+## Abstract
+
+Enable effective prioritized transaction submission based on gas price.
+
+## Motivation
+
+Currently, there is no deterministic gas ordering for the transactions, even when these transactions are submitted within an extremely short time window (less than 100ms). This is due to the fact that different transactions are randomly submitted by different validators, and the validator that is supposed to submit the transaction can have latency jitters. We propose a mechanism to mitigate the random factor and enable prioritized transaction submission.
+
+## Specification
+
+We propose the following changes:
+
+- Increase the maximum allowed value of gas price from $99,999$ to $1,000,000,000,000$ (1T).
+- Allow immediate transaction broadcast based on gas price.
+  - Once a validator receives a transaction with a higher gas price, it is submitted immediately to consensus regardless of tx digest, with the least possible delay, if the validator is in the range of $[0, N)$ of submission position.
+  - $N$ is a function of gas price, defined as:
+
+    - if $`gas\_price < K \times RGP`$ then $N = 1$
+    - else $`N = gas\_price / RGP + 1`$
+
+We acknowledge that in order to adapt to the forthcoming Mysticeti Fast Path, further changes are needed.  
+
+$K$ is a constant chosen to limit amplification. A smaller value of $K$ will increase amplification to the Sui network. In this case, $5$ is chosen as the initial value of $K$.
+
+## Rationale
+
+Transaction ordering on Sui depends on many factors that may or may not be within the control of the user.
+
+Consider such a scenario where multiple actors are competing against one trading opportunity, where which transaction lands first wins. One actor may send a transaction, specifying the maximum possible gas price (99,999), hoping that the transaction will be included in the earliest possible position in the very next round of consensus commits. However, doing so will not reliably succeed. In fact, she may find out that despite paying the highest possible gas price, transaction inclusion still appears random, the effect of using a higher gas price is negligible.
+
+We have located and analyzed multiple CEX-DEX trading bots and found no exception other than using a constant gas price (for example, 850 MIST), despite the fact that they are competing with each other.
+
+As of today, there is little one can do about the case, other than submitting many different transactions that are further submitted by different validators, or simply resort to luck.
```

Comment:
Another way to guarantee fast transaction submission is for validators to bypass latency jitter. Afaik, jitter isn’t verifiable or enforced—validators can take off-chain tips and prioritize transactions. With this SIP, while off-chain prioritization can still occur, transaction reordering (post-consensus) ensures that gas price determines final order. This mitigates the off-chain fee market by prioritizing those who pay the correct gas price.

Reply:
Yes.  The choice of validator also matters in this off-chain fee market case. 
For example, cooperating with a validator in Asia would unlikely lead to desired result.  
Some other factors may also have impact on this, such as leader schedule and the timing of block proposing.